sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesCredential.cs

```diff
@@ -151,7 +151,10 @@ internal string GetOidcTokenResponse(HttpMessage message)
                 string error = $"OIDC token not found in response. " + Troubleshooting;
                 if (message.Response.Status != 200)
                 {
-                    error = error + $"\n\nResponse= {message.Response.Content}";
+                    var is_x_vss_e2eidValueFound = message.Response.Headers.TryGetValue("x-vss-e2eid", out var x_vss_e2eidValue);
+                    var is_x_msedge_refValueFound = message.Response.Headers.TryGetValue("x-msedge-ref", out var x_msedge_refValue);
+
+                    error = error + $"\n\nResponse= {message.Response.Content}\n\nx-vss-e2eid= {(is_x_vss_e2eidValueFound ? x_vss_e2eidValue : "Not Found")}\n\nx-msedge-ref= {(is_x_msedge_refValueFound ? x_msedge_refValue : "Not Found")}";
```

Comment:
This adds the headers to the exception message, as expected.

Add the headers to logging too? I imagine you'd need to do that by updating the ClientOptions passed to the http pipeline that handles the request->response to include the two headers in the allow list.

Reply:
My thought is that this log by default as part of the exception, so the additional logging info is not needed globally.